internal/servers/controller/controller.go

```diff
@@ -220,7 +220,21 @@ func New(ctx context.Context, conf *Config) (*Controller, error) {
 	); err != nil {
 		return nil, fmt.Errorf("error adding config keys to kms: %w", err)
 	}
-	if err := c.kms.ReconcileKeys(ctx, c.conf.SecureRandomReader); err != nil {
+
+	// we need to get all the scopes so we can reconcile the DEKs for each scope.
+	iamRepo, err := iam.NewRepository(dbase, dbase, c.kms, iam.WithRandomReader(c.conf.SecureRandomReader))
+	if err != nil {
+		return nil, fmt.Errorf("unable to initialize iam repository: %w", err)
+	}
+	allScopes, err := iamRepo.ListScopesRecursively(ctx, scope.Global.String())
+	if err != nil {
+		return nil, fmt.Errorf("error listing all scopes for reconciling keys: %w", err)
+	}
+	reconcileScopeIds := make([]string, 0, len(allScopes))
+	for _, s := range allScopes {
+		reconcileScopeIds = append(reconcileScopeIds, s.PublicId)
+	}
+	if err := c.kms.ReconcileKeys(ctx, c.conf.SecureRandomReader, kms.WithScopeIds(reconcileScopeIds...)); err != nil {
```

Comment:
This will turn into potentially a lot of DB calls (depending on the number of scopes) because we'll not have a warmed KMS cache at this point and it's going to fetch roots for all scopes. Doesn't need to be addressed now but should be in a follow up.

Reply:
This will be addressed in a soonish PR that ports boundary over to use the new go-kms-wrapping/extras/kms package.